packages/create-plugin/src/commands/generate.plopfile.ts

```diff
@@ -113,6 +113,22 @@ export default function (plop: NodePlopAPI) {
         ? getActionsForTemplateFolder({ folderPath: backendFolderPath, exportPath })
         : [];
 
+      // Common, pluginType and backend actions can contain the same destination filenames.
+      // This filtering removes the duplicates to prevent plop erroring and makes sure the correct
+      // template is scaffolded.
+      const pluginActions = [...commonActions, ...pluginTypeSpecificActions, ...backendActions]
+        .reduceRight((acc, file) => {
+          const exists = acc.some((f) => f.path === file.path);
+          // if the file.path already exists and isn't a modification exclude it.
+          if (exists && file.type !== 'modify') {
```

Comment:
What happens if the existing file also has `file.type = 'modify'`? Are we overriding it in that case, and not keeping the precedence of the last action?

Reply:
(After talking to Jack I actually understand now, resolving.)

Reply:
We're looking to only exclude additional entries in the array that have `file.type ==='add'` if there is already an entry with `file.type === 'add'` and the file.paths match. I've changed the logic and the comment to try to make it easier to reason with.

Reply:
I'm also wondering if we should extract the whole function and write some tests to assert these things are doing their intended job! 😄

Reply:
Hehe, we could do that, too, however maybe in this case it would be better to write some "E2E" tests for the scaffolding scenarios, and have a case for testing this functionality as well? (This of course doesn't mean that we couldn't / shouldn't have tests for the units themselves)